marine/marine.py

```diff
@@ -154,3 +175,37 @@ def _add_or_get_filter(
 
     def __del__(self):
         self._marine.destroy_marine()
+
+    @staticmethod
```

Comment:
`classmethod`s are preferred.
They are more explicitly overridable and provide you an often needed context (`cls`).